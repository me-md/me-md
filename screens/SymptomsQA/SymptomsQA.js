import React, { useState, useEffect } from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { Entypo } from '@expo/vector-icons';
import { LinearGradient } from 'expo-linear-gradient';
import { ScrollView } from 'react-native-gesture-handler';
import SingleQ from '../../components/SingleQ';
import GroupSingleQ from '../../components/GroupSingleQ';
import { sendInitialUserSymptoms } from '../../utils/apiCalls/apiCalls';

const height = Dimensions.get('window').height;

export default function SymptomsQA({ navigation }) {
  // const { age, location, presentFactors, sex, symptomFollowup } = navigation.state.params;
  // const [currentQuestion, setCurrentQuestion] = useState({})
  const { cleanedData, location, stateAbbreviation } = navigation.state.params;
  const [question, setQuestion] = useState({});
  const [userInfo, setUserInfo] = useState(cleanedData);

  useEffect(() => {
    getQA();
  }, [userInfo]);

  const getQA = async () => {
    try {
      let symptomFollowup = await sendInitialUserSymptoms(userInfo);
      setQuestion(symptomFollowup);
      symptomFollowup.should_stop
        ? navigation.navigate('Results', {
            userInfo,
            symptomFollowup,
            location,
            stateAbbreviation
          })
        : null;
    } catch (error) {
      throw new Error(error);
    }
  };

  const displayQuestion = () => {
    switch (question.question.type) {
      case 'single':
        return (
          <SingleQ
            style={styles.question}
            question={question}
            answerQuestion={answerQuestion}
          />
        );
      case 'group_single':
        return (
          <GroupSingleQ
            style={styles.question}
            question={question}
            answerQuestion={answerQuestion}
          />
        );
      case 'group_multiple':
        return (
          <GroupSingleQ
            style={styles.question}
            question={question}
            answerQuestion={answerQuestion}
          />
        );
      default:
        <></>;
    }
  };

  const answerQuestion = answers => {
    const newUserInfo = { ...userInfo };
    newUserInfo.evidence.push(...answers);
    setUserInfo(newUserInfo);
  };

  return (
    <View style={styles.container}>
      <LinearGradient
        style={styles.gradientBackground}
        colors={['#004EFF', '#88CCF1']}
      >
        <View style={styles.contentContainer}>
          <Entypo
            name='chevron-thin-up'
            size={36}
            color='white'
            onPress={() => navigation.navigate('SearchSymptoms')}
          />
          <Text style={styles.symptomText}>Regarding your symptoms:</Text>
          <ScrollView style={styles.card}>
            {question.question && displayQuestion()}
          </ScrollView>
          <Entypo
            name='chevron-thin-down'
            size={36}
            color='white'
            // onPress={() => }
          />
        </View>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  gradientBackground: {
    alignItems: 'center',
    flex: 1
  },
  container: {
    flex: 1,
    width: '100%'
  },
  contentContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    marginBottom: height * 0.05,
    marginTop: height * 0.05,
    width: '100%'
  },
  symptomText: {
    alignSelf: 'flex-start',
    fontSize: 16,
    padding: 0
  },
  card: {
    flex: 1,
    padding: 0,
    width: '100%'
  },
  question: {
    width: '80%'
  }
});
