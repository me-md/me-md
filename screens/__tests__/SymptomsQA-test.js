import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import SymptomsQA from '../../screens/SymptomsQA/SymptomsQA';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading'
}));

describe('SymptomsQA', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const mockNavigation = {
      state: {
        params: {
          cleanedData: {},
          stateAbbreviation: 'CO',
          location: {
            lat: 121,
            long: 25
          }
        }
      }
    };

    // cleanedData, location, stateAbbreviation

    const tree = renderer
      .create(<SymptomsQA navigation={mockNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
