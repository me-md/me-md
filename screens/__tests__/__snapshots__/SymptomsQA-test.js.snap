// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymptomsQA should match a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginBottom": 66.7,
          "marginTop": 66.7,
          "width": "100%",
        }
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": 26.68,
              "padding": 0,
            },
          ]
        }
        uppercase={false}
      >
        Regarding your symptoms
      </Text>
      <RCTScrollView
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        style={
          Object {
            "padding": 0,
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
